modules/installation-azure-marketplace-subscribe.adoc

```diff
@@ -0,0 +1,177 @@
+----
+--
+
++
+[NOTE]
+====
+Regardless of the version of {product-title} you are installing, the correct version of the Azure Marketplace image to use is 4.8.x. If required, as part of the installation process, your VMs are automatically upgraded.
+====
+. Inspect the image for your offer by running one of the following commands:
+** North America:
++
+[source,terminal]
+----
+$ az vm image show --urn redhat:rh-ocp-worker:rh-ocp-worker:<version>
+----
+** EMEA:
++
+[source,terminal]
+----
+$ az vm image show --urn redhat-limited:rh-ocp-worker:rh-ocp-worker:<version>
+----
+. Review the terms of the offer by running one of the following commands:
+** North America:
++
+[source,terminal]
+----
+$ az vm image terms show --urn redhat:rh-ocp-worker:rh-ocp-worker:<version>
+----
+** EMEA:
++
+[source,terminal]
+----
+$ az vm image terms show --urn redhat-limited:rh-ocp-worker:rh-ocp-worker:<version>
+----
+. Accept the terms of the offering by running one of the following commands:
+** North America:
++
+[source,terminal]
+----
+$ az vm image terms accept --urn redhat:rh-ocp-worker:rh-ocp-worker:<version>
+----
+** EMEA:
++
+[source,terminal]
+----
+$ az vm image terms accept --urn redhat-limited:rh-ocp-worker:rh-ocp-worker:<version>
+----
+ifdef::ipi[]
+. Record the image details of your offer. You must update the `compute` section in the `install-config.yaml` file with values for `publisher`, `offer`, `sku`, and `version` before deploying the cluster.
+endif::ipi[]
+ifdef::upi[]
+. Record the image details of your offer. If you use the Azure Resource Manager (ARM) template to deploy your worker nodes, you can update `storageProfile.imageReference` by deleting the `id` parameter and adding the `offer`, `publisher`, `sku`, and `version` parameters using the values from your offer.
+endif::upi[]
+ifdef::mapi[]
+. Record the image details of your offer, specifically the values for `publisher`, `offer`, `sku`, and `version`.
+endif::mapi[]
+
+ifdef::ipi[]
+.Sample `install-config.yaml` file with the Azure Marketplace worker nodes
+
+[source,yaml]
+----
+apiVersion: v1
+baseDomain: example.com
+compute:
+- hyperthreading: Enabled
+  name: worker
+  platform:
+    azure:
+      type: Standard_D4s_v5
+      osImage:
+        publisher: redhat
+        offer: rh-ocp-worker
+        sku: rh-ocp-worker
+        version: 4.8.2021122100
+  replicas: 3
+----
+endif::ipi[]
+ifdef::mapi[]
+. Add the following details to the `providerSpec` section of your machine set YAML file:
++
+.Sample `providerSpec` image values for Azure Marketplace compute machines
+
+[source,yaml]
+----
+providerSpec:
+        value:
+          image:
+            offer: rh-ocp-worker
+            publisher: redhat
+            resourceID: /resourceGroups/<infrastructure_id>-rg/providers/Microsoft.Compute/images/<infrastructure_id> <1>
+            sku: rh-ocp-worker
+            version: 4.8.2021122100
```

Comment:
this should be like below, we need to unset the `resourceID`, add `type: MarketplaceWithPlan`
```
          image:
            offer: rh-ocp-worker
            publisher: redhat
            resourceID: ""
            sku: rh-ocp-worker
            type: MarketplaceWithPlan
            version: 4.8.2021122100
```

Reply:
Thanks Zhaohua! I have updated to match your snippet. 

Just to make sure I understand: I have also removed the following note about the `resourceID`, is that correct?
```
<1> Specify an image that is compatible with your instance type. The Hyper-V generation V2 images created by the installation program have a `-gen2` suffix, while V1 images have the same name without the suffix.
```

Reply:
Yes, it is correct.

Reply:
Thank you, as always!